support digest in base image

spd-intermodalics opened the pull request:
# Problem description

[This patch](https://github.com/osrf/rocker/commit/80602eb6868a9044279d88c13650828f654a32f9) was breaking the ability of rocker to support base names that contained a `@` character as used for docker digests, or `.` needed for addresses of other than default registries.

## Fix

This PR adds support of digest as the base image name by also substituting `@` and `.` characters by underscores.

spd-intermodalics committed:
support digest in base image

This patch adds support of digest as the base image name by also
substituting @ characters by underscores.

spd-intermodalics committed:
support registry on base image

This patch fixes the support for base image that uses different
docker registry than the default.

meyerj commented:
I suggest to omit or revert the problematic patch instead in your merge or cherry-pick, because it has other negative consequences (https://github.com/osrf/rocker/pull/133#issuecomment-941622117). Temporary images should not have a tag assigned at all.

spd-intermodalics committed:
bugfix: Limit the tag length to 128 characters for OS detection image

Docker specifies that the tag part of the name (IMAGE[:TAG]) on an image
should contain a maximum of 128 characters. This patch enforces that for
the OS detection image.

meyerj commented:
Closing in favor of #7.
